superset-frontend/src/dashboard/components/Header/HeaderActionsDropdown/HeaderActionsDropdown.test.tsx

```diff
@@ -112,11 +116,84 @@ test('should render the menu items in edit mode', async () => {
   setup(editModeOnProps);
   expect(screen.getAllByRole('menuitem')).toHaveLength(4);
   expect(screen.getByText('Set auto-refresh interval')).toBeInTheDocument();
-  expect(screen.getByText('Set filter mapping')).toBeInTheDocument();
   expect(screen.getByText('Edit properties')).toBeInTheDocument();
   expect(screen.getByText('Edit CSS')).toBeInTheDocument();
 });
 
+describe('with native filters feature flag disabled', () => {
+  beforeAll(() => {
+    isFeatureEnabledMock = jest
+      .spyOn(featureFlags, 'isFeatureEnabled')
+      .mockImplementation(
+        (featureFlag: FeatureFlag) =>
+          featureFlag !== FeatureFlag.DASHBOARD_NATIVE_FILTERS,
+      );
+  });
+
+  afterAll(() => {
+    // @ts-ignore
+    isFeatureEnabledMock.restore();
+  });
+
+  it('should render filter mapping in edit mode if explicit filter scopes undefined', async () => {
+    setup(editModeOnProps);
+    expect(screen.getByText('Set filter mapping')).toBeInTheDocument();
+  });
+
+  it('should render filter mapping in edit mode if explicit filter scopes defined', async () => {
+    const mockedProps = {
```

Comment:
Nit: Can you extract `mockedProps` to a sharable constant similar to `editModeOnProps`? It could be something like `propsWithFilterScopes`